### PR TITLE
Task/cit 11

### PR DESCRIPTION
**Pull request by raghavkuk**

### CODE REVIEW

**Comment by jdamerow** [comment]
please resolve conflicts

File: citesphere/src/main/java/edu/asu/diging/citesphere/web/user/EditItemController.java [hunk 1]
```diff
@@ -25,161 +29,302 @@
...
+            System.out.println(citation.getAuthors().size());
+            citation.getEditors().forEach(e -> e.getAffiliations().size());
+            citationManager.detachCitation(citation);
+            citationHelper.updateCitation(citation, form);
+
+            try {
+                citationManager.updateCitation((IUser) authentication.getPrincipal(), zoteroGroupId, citation);
+            } catch (CitationIsOutdatedException e) {
+                System.out.println(e);
+            }
```
Comment (jdamerow) [inline citesphere/src/main/java/edu/asu/diging/citesphere/web/user/EditItemController.java]: this is not how we handle expceptions

Comment (jdamerow) [reply citesphere/src/main/java/edu/asu/diging/citesphere/web/user/EditItemController.java]: also, in this case we want to show the resolve conflict page as we did before

File: citesphere/src/main/java/edu/asu/diging/citesphere/web/user/EditItemController.java [hunk 2]
```diff
@@ -25,161 +29,302 @@
...
-                citationField.setAccessible(true);
-                String citationValue = citationField.get(citation) != null ? citationField.get(citation).toString() : "";
-                // if values are the same, continue
-                formValue = formValue != null ? formValue : "";
-                citationValue = citationValue != null ? citationValue : "";
-                
-                if (!formValue.equals(citationValue)) {
-                    changedFields.add(fieldname);                
+
+                if (fieldname.equals("authors") || fieldname.equals("editors")) {
```
Comment (jdamerow) [inline citesphere/src/main/java/edu/asu/diging/citesphere/web/user/EditItemController.java]: what about otherCreators?

File: citesphere/src/main/java/edu/asu/diging/citesphere/web/user/EditItemController.java [hunk 3]
```diff
@@ -25,161 +29,302 @@
...
+                                ? citationField.get(citation).toString()
+                                : "";
+                        field.setAccessible(true);
+                        field.set(form, citationValue);
+                    } catch (NoSuchFieldException | SecurityException | IllegalArgumentException
+                            | IllegalAccessException e) {
+
+                        logger.error("Could ont access field.", e);
+                        // let's ignore that for nonw
+                    }
```
Comment (jdamerow) [inline citesphere/src/main/java/edu/asu/diging/citesphere/web/user/EditItemController.java]: same here, what about otherCreators?

File: citesphere/src/main/webapp/WEB-INF/views/auth/group/editConflict.jsp [hunk 4]
```diff
@@ -5,6 +5,71 @@
...
+				
+				var affiliationIdField = $("<input>");
+				affiliationIdField.attr("type", "hidden");
+				affiliationIdField.attr("id", arrayName+"s" + idx + ".affiliations" + idx2 + ".id");
+				affiliationIdField.attr("name", arrayName+"s[" + idx + "].affiliations[" + idx2 + "].id");
+				affiliationIdField.attr("value", $(affiliation).attr("data-affiliation-id"));
+				$("#conflictForm").append(affiliationIdField);
+			});
+	});
+}
```
Comment (jdamerow) [inline citesphere/src/main/webapp/WEB-INF/views/auth/group/editConflict.jsp]: identation

File: citesphere/src/main/java/edu/asu/diging/citesphere/web/user/EditItemController.java [hunk 5]
```diff
@@ -25,161 +29,302 @@
...
+    @RequestMapping(value = "/auth/group/{zoteroGroupId}/items/{itemId}/edit", method = RequestMethod.POST)
+    public String storeItem(@ModelAttribute CitationForm form, Authentication authentication, Model model,
+            @PathVariable("zoteroGroupId") String zoteroGroupId, @PathVariable("itemId") String itemId)
+            throws ZoteroConnectionException, GroupDoesNotExistException {
+        ICitation citation = citationManager.getCitation((IUser) authentication.getPrincipal(), zoteroGroupId, itemId);
+        if (form.getVersion() == citation.getVersion()) {
+
+            // load authors and editors before detaching
+            citation.getAuthors().forEach(a -> a.getAffiliations().size());
+            System.out.println(citation.getAuthors().size());
```
Comment (jdamerow) [inline citesphere/src/main/java/edu/asu/diging/citesphere/web/user/EditItemController.java]: remove

File: citesphere/src/main/java/edu/asu/diging/citesphere/web/user/EditItemController.java [hunk 6]
```diff
@@ -25,161 +29,302 @@
...
+            citationHelper.updateCitation(citation, form);
+
+            try {
+                citationManager.updateCitation((IUser) authentication.getPrincipal(), zoteroGroupId, citation);
+            } catch (CitationIsOutdatedException e) {
+                System.out.println(e);
+            }
+
+        } else {
+            citation = citationManager.getCitation((IUser) authentication.getPrincipal(), zoteroGroupId, itemId);
```
Comment (jdamerow) [inline citesphere/src/main/java/edu/asu/diging/citesphere/web/user/EditItemController.java]: this is not used anymore, is it? remove

File: citesphere/src/main/java/edu/asu/diging/citesphere/web/user/EditItemController.java (hunk 1 above)
Comment (raghavkuk) [reply citesphere/src/main/java/edu/asu/diging/citesphere/web/user/EditItemController.java]: The resolve conflict page is being handled by the else condition. An exception here would simply result in a failed update.

**Pull request closed without merging**